### PR TITLE
894 svg icon template

### PR DESCRIPTION
**Pull request by cmdcolin**
This goes some way towards the SVG icon issue

Addresses #894

### CODE REVIEW

**Comment by cmdcolin** [comment]
This seems like it's going ok but the draggable track handles don't work. Not sure why... @garrettjstevens

File: packages/core/ui/ViewContainer.tsx [hunk 1]
```diff
@@ -208,9 +199,7 @@ export default withContentRect('bounds')(
               edge="end"
               onClick={onClose}
             >
-              <Icon fontSize="small" className={classes.icon}>
-                close
-              </Icon>
+              <CloseIcon className={classes.icon} />
```
Comment (garrettjstevens) [inline packages/core/ui/ViewContainer.tsx:202]: missing fontSize="small" here

File: packages/linear-genome-view/src/LinearGenomeView/components/TrackLabel.tsx [hunk 2]
```diff
@@ -113,23 +117,22 @@ const TrackLabel = React.forwardRef(
...
             className={classes.dragHandle}
-            fontSize="small"
             onDragStart={onDragStart}
             onDragEnd={onDragEnd}
             data-testid={`dragHandle-${view.id}-${getConf(track, 'trackId')}`}
           >
-            drag_indicator
-          </Icon>
+            <DragIcon fontSize="small" />
+          </div>
```
Comment (garrettjstevens) [inline packages/linear-genome-view/src/LinearGenomeView/components/TrackLabel.tsx:128]: This should fix the track dragging:

```diff
--- a/packages/linear-genome-view/src/LinearGenomeView/components/TrackLabel.tsx
+++ b/packages/linear-genome-view/src/LinearGenomeView/components/TrackLabel.tsx
@@ -34,11 +34,14 @@ const useStyles = makeStyles(theme => ({
     pointerEvents: 'none',
   },
   dragHandle: {
-    display: 'inline',
-    verticalAlign: 'middle',
     cursor: 'grab',
     color: '#135560',
   },
+  dragHandleIcon: {
+    display: 'inline-block',
+    verticalAlign: 'middle',
+    pointerEvents: 'none',
+  },
   iconButton: {
     padding: theme.spacing(1),
   },
@@ -117,15 +120,15 @@ const TrackLabel = React.forwardRef(
     return (
       <>
         <Paper ref={ref} className={clsx(className, classes.root)}>
-          <div
+          <span
             draggable
             className={classes.dragHandle}
             onDragStart={onDragStart}
             onDragEnd={onDragEnd}
             data-testid={`dragHandle-${view.id}-${getConf(track, 'trackId')}`}
           >
-            <DragIcon fontSize="small" />
-          </div>
+            <DragIcon fontSize="small" className={classes.dragHandleIcon} />
+          </span>
           <IconButton
             onClick={() => view.hideTrack(track.configuration)}
             className={classes.iconButton}
```

File: packages/spreadsheet-view/src/SpreadsheetView/components/Spreadsheet.js [hunk 3]
```diff
@@ -1,3 +1,8 @@
+import ArrowUp from '@material-ui/icons/ArrowUpward'
+import ArrowDown from '@material-ui/icons/ArrowDownward'
```
Comment (garrettjstevens) [inline packages/spreadsheet-view/src/SpreadsheetView/components/Spreadsheet.js:2]: Should be keyboard arrows
```suggestion
import KeyboardArrowUpIcon from '@material-ui/icons/KeyboardArrowUp'
import KeyboardArrowDownIcon from '@material-ui/icons/KeyboardArrowDown'
```

File: packages/spreadsheet-view/src/SpreadsheetView/components/Spreadsheet.js [hunk 4]
```diff
@@ -195,13 +198,9 @@ export default pluginManager => {
...
-        <Icon
-          fontSize="small"
-          className={classes.sortIndicator}
-          onClick={sortSpec.switchDirection}
-        >
-          {descending ? 'keyboard_arrow_up' : 'keyboard_arrow_down'}
-        </Icon>
+        <div fontSize="small" className={classes.sortIndicator}>
+          {descending ? <ArrowUp /> : <ArrowDown />}
+        </div>
```
Comment (garrettjstevens) [inline packages/spreadsheet-view/src/SpreadsheetView/components/Spreadsheet.js:203]: fontSize and className on the containing div doesn't seem to be working
```suggestion
        descending ? (
          <KeyboardArrowUpIcon
            fontSize="small"
            className={classes.sortIndicator}
          />
        ) : (
          <KeyboardArrowDownIcon
            fontSize="small"
            className={classes.sortIndicator}
          />
        )
```

File: packages/spreadsheet-view/src/SpreadsheetView/components/SpreadsheetView.js [hunk 5]
```diff
@@ -83,7 +84,7 @@ export default pluginManager => {
             data-testid="spreadsheet_view_open"
             color="secondary"
           >
-            <Icon fontSize="small">folder_open</Icon>
+            <FolderOpenIcon />
```
Comment (garrettjstevens) [inline packages/spreadsheet-view/src/SpreadsheetView/components/SpreadsheetView.js:87]: missing fontSize="small" here

File: packages/dotplot-view/src/DotplotView/components/Controls.tsx [hunk 6]
```diff
@@ -1,4 +1,11 @@
 import { makeStyles as makeStylesMUI } from '@material-ui/core/styles'
+import ZoomOut from '@material-ui/icons/ZoomOut'
+import ZoomIn from '@material-ui/icons/ZoomIn'
+import ArrowUp from '@material-ui/icons/ArrowUpward'
+import ArrowDown from '@material-ui/icons/ArrowDownward'
+import ArrowLeft from '@material-ui/icons/ArrowLeft'
+import ArrowRight from '@material-ui/icons/ArrowRight'
```
Comment (garrettjstevens) [inline packages/dotplot-view/src/DotplotView/components/Controls.tsx:7]: Should be keyboard arrow icons

File: packages/dotplot-view/src/DotplotView/components/Controls.tsx [hunk 7]
```diff
@@ -45,7 +51,7 @@ export default (pluginManager: any) => {
           title="left"
           color="secondary"
         >
-          <Icon fontSize="small">keyboard_arrow_left</Icon>
+          <ArrowLeft />
```
Comment (garrettjstevens) [inline packages/dotplot-view/src/DotplotView/components/Controls.tsx:54]: missing fontSize="small" (and others below)

File: packages/spreadsheet-view/src/SpreadsheetView/components/ColumnFilterControls.js [hunk 8]
```diff
@@ -62,17 +64,15 @@ export default pluginManager => {
           style={{ height }}
         >
           <Grid item className={classes.filterIconBg}>
-            <Icon className={classes.filterIcon} fontSize="small">
-              filter_list
-            </Icon>
+            <FilterIcon className={classes.filterIcon} />
```
Comment (garrettjstevens) [inline packages/spreadsheet-view/src/SpreadsheetView/components/ColumnFilterControls.js:67]: missing fontSize="small"

File: packages/spreadsheet-view/src/SpreadsheetView/components/ColumnFilterControls.js [hunk 9]
```diff
@@ -62,17 +64,15 @@ export default pluginManager => {
...
           </Grid>
           <Grid item>
             <IconButton
               onClick={removeFilter}
               title="remove filter"
               color="secondary"
             >
-              <Icon fontSize="small">close</Icon>
+              <CloseIcon />
```
Comment (garrettjstevens) [inline packages/spreadsheet-view/src/SpreadsheetView/components/ColumnFilterControls.js:75]: missing fontSize="small"

File: packages/sv-inspector/src/SvInspectorView/components/SvInspectorView.js [hunk 10]
```diff
@@ -69,7 +70,7 @@ export default pluginManager => {
             data-testid="sv_inspector_view_open"
             color="secondary"
           >
-            <Icon fontSize="small">folder_open</Icon>
+            <FolderOpenIcon />
```
Comment (garrettjstevens) [inline packages/sv-inspector/src/SvInspectorView/components/SvInspectorView.js:73]: missing fontSize="small"

File: packages/linear-genome-view/src/LinearGenomeView/components/TrackLabel.tsx (hunk 2 above)
Comment (cmdcolin) [reply packages/linear-genome-view/src/LinearGenomeView/components/TrackLabel.tsx]: nice :) looks like that does it

File: packages/core/ui/ViewContainer.tsx [hunk 11]
```diff
@@ -73,30 +73,14 @@ const useStyles = makeStyles(theme => ({
 const ViewMenu = observer(
   ({
     model,
-    IconButtonProps,
-    IconProps,
+    iconButtonProps,
```
[the remark this answers is not in the document]
Comment (cmdcolin) [reply packages/core/ui/ViewContainer.tsx:76]: i'll reset but the name collision of the value, IconButtonProps, colliding with the type, IconButtonProps, seemed sort of tedious. I guess this is just something import renaming will deal with

**Comment by cmdcolin** [comment]
RE: appending all icon names with the word Icon, I am personally ok with a little brevity to save my fingers

**Comment by cmdcolin** [comment]
Should have all the suggested fixes now

**Comment by rbuels** [comment]
you guys can merge this whenever you both feel it's good

**Pull request merged**